### PR TITLE
Replace deprecated set-output commands

### PR DESCRIPTION
**Pull request by aegjoyce**
Updates builder to use the new github environment files for processing workflows. Also add package-writing permissions so the default github token can be used without errors

### CODE REVIEW

File: .github/workflows/builder.yaml [hunk 1]
```diff
@@ -12,6 +12,10 @@ on:
     branches:
       - main
 
+permissions:
+  contents: read
+  packages: write
```
Comment (frenck) [inline .github/workflows/builder.yaml:17]: This is not what the title of the PR said, maybe make this a separate PR? Additionally, it might be better to only pass write permissions to the specific tasks that need those (instead of the whole job).

**Review by frenck — CHANGES_REQUESTED** [review]

File: .github/workflows/builder.yaml (hunk 1 above)
Comment (aegjoyce) [reply .github/workflows/builder.yaml]: Apologies, it snuck in with the other changes - I have removed it

**Comment by aegjoyce** [comment]
Removed changes related to permission that were not part of the scope of this PR

**Review by frenck — APPROVED** [review]
Thanks, @aegjoyce 👍 

../Frenck
